### PR TITLE
Fixes for log tailing

### PR DESCRIPTION
**Pull request by timstclair**
IN_ATTRIB inotify events are generated when atime / mtime is changed,
which would cause the tail to be reset, and reread the same log
again (generating duplicate events). Instead, watch the directory for
file delete / move.

Also, use an exponential backoff when retrying opening the file.

/cc @sjenning

### CODE REVIEW

**Comment by sjenning** [comment]
wfm and lgtm, thanks

File: utils/tail/tail.go [hunk 1]
```diff
@@ -105,8 +92,11 @@ func (t *Tail) attemptOpen() error {
 			t.reader = bufio.NewReader(t.file)
```
Comment (vishh) [inline utils/tail/tail.go]: Why is the line above commented?

File: utils/tail/tail.go [hunk 2]
```diff
@@ -133,15 +123,24 @@ func (t *Tail) watchFile() error {
 		return err
 	}
 	defer t.file.Close()
-	err = t.watcher.Watch(t.filename)
+
+	watchDir := filepath.Dir(t.filename)
+	err = t.watcher.AddWatch(watchDir, inotify.IN_MOVED_FROM|inotify.IN_DELETE)
```
Comment (vishh) [inline utils/tail/tail.go]: Are we sure that a rename is what logrotation does? Sounds most likely, but just checking..

**Comment by vishh** [comment]
LGTM

File: utils/tail/tail.go (hunk 1 above)
Comment (timstclair) [inline utils/tail/tail.go]: It's tied to the TODO. I actually don't think we want to do that...

File: utils/tail/tail.go (hunk 2 above)
Comment (timstclair) [inline utils/tail/tail.go]: Manually verified this works by forcing a logrotate.

**Pull request merged**